File: problems/basic/23/solution1.ts
// Accepted: 120ms (58.02%), 61.35MB (5.41%)

/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function mergeKLists(lists: Array<ListNode | null>): ListNode | null {
    const queue = new MinPriorityQueue();

    for (let node of lists) {
        while (node !== null) {
            queue.enqueue(node.val);
            node = node.next;
        }
    }

    const mergedHead = new ListNode();
    let cur = mergedHead;
    while (!queue.isEmpty()) {
        const val = queue.dequeue().element;
        cur.next = new ListNode(val);
        cur = cur.next;
    }

    return mergedHead.next;
};

File: problems/234/solution1.py
# Accepted: 307ms (90.79%), 37.37MB (81.55%)

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        numbers = []

        node = head
        while node is not None:
            numbers.append(node.val)
            node = node.next

        for i in range(len(numbers)):
            if numbers[i] != numbers[len(numbers)-1-i]:
                return False
        return True
